hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -602,6 +603,10 @@ void Vgen::emit(const andli& i) {
   a->And(W(i.d), W(i.s1), MSKTOP(i.s0.l()), SetFlags);
 }
 
+void Vgen::emit(const testbi& i) {
+  a->Tst(W(i.s1), MSKTOP(0xff));
```

Comment:
This is still wrong, in the way that @dave-estes mentioned.  As you have it, regardless of what you set `s0` to, it'll be essentially replaced with `0xff`.  You're basically making all of the following transformations:

```
testbi{0x0, r} -> testbi{0xff, r}
testbi{0x1, r} -> testbi{0xff, r}
testbi{0x2, r} -> testbi{0xff, r}
...
```

You're also still only masking the top 32 bits, instead of the top 56 bits as is necessary for correctness, in case there's junk in the upper 56 bits of `i.s1`.